### PR TITLE
Add clearer error message for map.queryRenderedFeatures...

### PR DESCRIPTION
**Pull request by mollymerp**
when called with a layer that does not exist in the map's style.
#3084 

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [ ] document any changes to public APIs n/a
 - [ ] post benchmark scores n/a
 - [x] manually test the debug page 
 - [ ] get a PR review

### CODE REVIEW

File: debug/index.html [hunk 1]
```diff
@@ -115,6 +115,7 @@
 });
 
 map.on('click', function(e) {
+    var features = map.queryRenderedFeatures(e.point, {layers:['derp']});
```
Comment (mourner) [inline debug/index.html]: Not necessary for this debug page.

File: js/style/style.js [hunk 2]
```diff
@@ -647,6 +647,10 @@ Style.prototype = util.inherit(Evented, {
         if (params && params.layers) {
             for (var i = 0; i < params.layers.length; i++) {
                 var layerId = params.layers[i];
+                if (!(this._layers[layerId] instanceof StyleLayer)) {
+                    // this layer is not in the style.layers array
+                    return this.fire('error', {error: 'The layer \'' + layerId + '\' does not exist in the map\'s style and cannot be queried for features.'});
+                }
                 includedSources[this._layers[layerId].source] = true;
```
Comment (mourner) [inline js/style/style.js]: You could remove some repetition here by doing:

```js
var layer = this._layers[params.layers[i]];
...
layer.id
```

File: test/js/style/style.test.js [hunk 3]
```diff
@@ -1093,6 +1093,16 @@ test('Style#queryRenderedFeatures', function(t) {
             t.end();
         });
 
+        t.test('fires an error if layer included in params does not exist on the style', function(t) {
+            var errors = 0;
+            sinon.stub(style, 'fire', function() {
+                if (arguments[1].error && arguments[1].error.includes('does not exist in the map\'s style and cannot be queried for features.')) errors++;
```
Comment (mourner) [inline test/js/style/style.test.js]: I think it's cleaner to use usual function parameters rather than `arguments`.

**Comment by mollymerp** [comment]
fixed @mourner !

File: js/style/style.js [hunk 4]
```diff
@@ -646,8 +646,12 @@ Style.prototype = util.inherit(Evented, {
         var includedSources = {};
         if (params && params.layers) {
             for (var i = 0; i < params.layers.length; i++) {
-                var layerId = params.layers[i];
-                includedSources[this._layers[layerId].source] = true;
+                var layer = this._layers[params.layers[i]];
+                if (!(layer instanceof StyleLayer)) {
+                    // this layer is not in the style.layers array
+                    return this.fire('error', {error: 'The layer \'' + params.layers[i] + '\' does not exist in the map\'s style and cannot be queried for features.'});
```
Comment (mourner) [inline js/style/style.js]: This line is a bit too long, can we make it two lines?

**Pull request merged**